engine/src/main/java/com/redhat/cloud/notifications/exports/ExportsEventListener.java

```diff
@@ -0,0 +1,335 @@
+import com.redhat.cloud.notifications.exports.transformers.event.JSONEventTransformer;
+import com.redhat.cloud.notifications.models.Event;
+import io.micrometer.core.instrument.MeterRegistry;
+import io.quarkus.logging.Log;
+import io.smallrye.common.annotation.Blocking;
+import org.apache.http.HttpStatus;
+import org.eclipse.microprofile.config.inject.ConfigProperty;
+import org.eclipse.microprofile.reactive.messaging.Incoming;
+import org.eclipse.microprofile.reactive.messaging.Message;
+import org.eclipse.microprofile.rest.client.inject.RestClient;
+
+import javax.enterprise.context.ApplicationScoped;
+import javax.inject.Inject;
+import javax.transaction.Transactional;
+import java.time.LocalDate;
+import java.time.ZoneOffset;
+import java.time.format.DateTimeParseException;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.UUID;
+import java.util.concurrent.CompletionStage;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+@ApplicationScoped
+public class ExportsEventListener {
+
+    public static final String APPLICATION_NAME = "urn:redhat:application:notifications";
+    public static final String CE_EXPORT_REQUEST_TYPE = "com.redhat.console.export-service.request";
+    public static final String EXPORT_CHANNEL = "export-requests";
+    public static final String EXPORT_SERVICE_URN = "urn:redhat:source:console:app:export-service";
+    public static final String FILTER_DATE_FROM = "from";
+    public static final String FILTER_DATE_TO = "to";
+    public static final String RESOURCE_TYPE_EVENTS = "urn:redhat:application:notifications:export:events";
+
+    protected static final String EXPORTS_SERVICE_FAILURES_COUNTER = "exports.service.failures";
+    protected static final String EXPORTS_SERVICE_SUCCESSES_COUNTER = "exports.service.successes";
+
+    private final ConsoleCloudEventParser consoleCloudEventParser = new ConsoleCloudEventParser();
+    private final Pattern subjectUuidExtractPattern = Pattern.compile("^urn:redhat:subject:export-service:request:(?<uuid>.+)$");
+
+    @Inject
+    EventRepository eventRepository;
+
+    @RestClient
+    ExportsService exportsService;
+
+    @Inject
+    FeatureFlipper featureFlipper;
+
+    @Inject
+    MeterRegistry meterRegistry;
+
+    @Inject
+    StatelessSessionFactory statelessSessionFactory;
+
+    @ConfigProperty(name = "export-service.psk")
+    String exportServicePsk;
+
+    /**
+     * Listens to the exports channel, extracts the request's data, and if
+     * Notifications is the target application, and the requested resource and
+     * formats are supported, the corresponding payload is sent to the export
+     * service.
+     * @param message the incoming message from the channel.
+     * @return a read commit on the received message.
+     */
+    @Blocking
+    @Incoming(EXPORT_CHANNEL)
+    @Transactional
+    public CompletionStage<Void> eventListener(final Message<String> message) {
```

Comment:
If you don't need to access the headers of the Kafka message, then this signature should be enough:
```suggestion
    public void eventListener(final String payload) {
```

Reply:
Then you can replace `return message.ack();` with `return;` everywhere.

Reply:
We'll simplify to that, for sure. Thanks!